Unable to build oneDNN using mingw32

# Summary
OneDNN doesn't build with mingw32

# Version

v2.3 (https://github.com/oneapi-src/oneDNN/releases/tag/v2.3)

# Environment
```
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
Address sizes:       46 bits physical, 48 bits virtual
CPU(s):              20
On-line CPU(s) list: 0-19
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               85
Model name:          Intel(R) Core(TM) i9-7900X CPU @ 3.30GHz
Stepping:            4
CPU MHz:             1921.675
CPU max MHz:         4500.0000
CPU min MHz:         1200.0000
BogoMIPS:            6599.98
Virtualization:      VT-x
NUMA node0 CPU(s):   0-19
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm co
                     nstant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fm
                     a cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_
                     l3 cdp_l3 invpcid_single pti ssbd mba ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rt
                     m cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_l
                     lc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req md_clear flush_l1d
```
* ```Linux 1642c05710cf 4.15.0-133-generic #137~16.04.1-Ubuntu SMP Fri Jan 15 02:55:18 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux```
* Fedora release 34 (Thirty Four) (container)
* Compiler version : x86_64-w64-mingw32-gcc (GCC) 10.2.1 20200723 (Fedora MinGW 10.2.1-6.fc34)
* CMake version : cmake version 3.20.5

# Steps to reproduce
Just try to build using x86_64-w64-mingw32-gcc.  Here's the cmake command I'm using

```
CC=x86_64-w64-mingw32-gcc
CXX=x86_64-w64-mingw32-g++

CXXFLAGS="-w -D_WIN32_WINNT=0x0600" CXXFLAGS="-w  -D_WIN32_WINNT=0x0600" cmake .. -DCMAKE_TOOLCHAIN_FILE=/usr/share/mingw/toolchain-mingw64.cmake -DCMAKE_CXX_FLAGS_RELEASE="-O2 -DNDEBUG" -DCMAKE_INSTALL_PREFIX=`realpath ../install` -DDNNL_CPU_RUNTIME="SEQ" -DDNNL_LIBRARY_TYPE="STATIC" -DDNNL_BUILD_EXAMPLES=OFF -DDNNL_ENABLE_JIT_PROFILING=OFF -DDNNL_BUILD_TESTS=OFF -DDNNL_VERBOSE=OFF -DDNNL_ENABLE_MAX_CPU_ISA=ON -DDNNL_ENABLE_CPU_ISA_HINTS=OFF 
```

I reproduced this using the v2.3 tag.  However, the issue is with the ittnotify library (see below) and I can see that it is still present on the tip of the master branch, i.e., the itnotify library has not been updated since v2.3 and so the issue must still be present.

# Observed behavior


```
In file included from oneDNN/src/common/ittnotify/ittnotify_static.c:61:
oneDNN/src/common/ittnotify/ittnotify_config.h:286:1: error: multiple storage classes in declaration specifiers
  286 | ITT_INLINE long
      | ^~~~~~~~~~
oneDNN/src/common/ittnotify/ittnotify_config.h:288:1: error: multiple storage classes in declaration specifiers
  288 | ITT_INLINE long __itt_interlocked_increment(volatile long* ptr)
      | ^~~~~~~~~~
...
```

The issue is with the vendored ittnotify component and has been fixed upstream.  The relevant commits seem to be

https://github.com/intel/ittapi/commit/87bafb65f6ee2fa654f4bd355d4e19e68363c7bc#diff-093a63068100815f13e2955a27c1830908aa09b7ea543b85d65a3f43ee793a47
https://github.com/intel/ittapi/commit/b1e7e75234929f75c8c3cbb2262ad9f287401115#diff-093a63068100815f13e2955a27c1830908aa09b7ea543b85d65a3f43ee793a47
https://github.com/intel/ittapi/commit/cad9530e4250c7caf4813f03d87e302115f6ca4a#diff-093a63068100815f13e2955a27c1830908aa09b7ea543b85d65a3f43ee793a47


# Expected behavior
I'd like to be able to build OneDNN cleanly with mingw32.  If I manually patch the vendored version of ittnotify using the commits linked to above, OneDNN builds fine under mingw32 and the resulting binary executes correctly on windows.

--- a/src/common/ittnotify/ittnotify.h
+++ b/src/common/ittnotify/ittnotify.h
@@ -240,7 +240,11 @@ The same ID may not be reused for different instances, unless a previous
 
 #if ITT_PLATFORM==ITT_PLATFORM_WIN
 /* use __forceinline (VC++ specific) */
+#if defined(__MINGW32__) && !defined(__cplusplus)
+#define ITT_INLINE           static __inline__ __attribute__((__always_inline__,__gnu_inline__))
+#else
 #define ITT_INLINE           static __forceinline
+#endif
 #define ITT_INLINE_ATTRIBUTE /* nothing */
 #else  /* ITT_PLATFORM==ITT_PLATFORM_WIN */
 /*

--- a/src/common/ittnotify/ittnotify_config.h
+++ b/src/common/ittnotify/ittnotify_config.h
@@ -153,7 +153,11 @@
 
 #if ITT_PLATFORM==ITT_PLATFORM_WIN
 /* use __forceinline (VC++ specific) */
+#if defined(__MINGW32__) && !defined(__cplusplus)
+#define ITT_INLINE           static __inline__ __attribute__((__always_inline__,__gnu_inline__))
+#else
 #define ITT_INLINE           static __forceinline
+#endif
 #define ITT_INLINE_ATTRIBUTE /* nothing */
 #else  /* ITT_PLATFORM==ITT_PLATFORM_WIN */
 /*

--- a/src/common/ittnotify/ittnotify_static.c
+++ b/src/common/ittnotify/ittnotify_static.c
@@ -61,7 +61,9 @@
 #include "ittnotify_config.h"
 
 #if ITT_PLATFORM==ITT_PLATFORM_WIN
+#if !defined(PATH_MAX)
 #define PATH_MAX 512
+#endif
 #else /* ITT_PLATFORM!=ITT_PLATFORM_WIN */
 #include <limits.h>
 #include <dlfcn.h>

--- a/src/common/ittnotify/legacy/ittnotify.h
+++ b/src/common/ittnotify/legacy/ittnotify.h
@@ -167,7 +167,11 @@
 
 #if ITT_PLATFORM==ITT_PLATFORM_WIN
 /* use __forceinline (VC++ specific) */
+#if defined(__MINGW32__) && !defined(__cplusplus)
+#define ITT_INLINE           static __inline__ __attribute__((__always_inline__,__gnu_inline__))
+#else
 #define ITT_INLINE           static __forceinline
+#endif
 #define ITT_INLINE_ATTRIBUTE /* nothing */
 #else  /* ITT_PLATFORM==ITT_PLATFORM_WIN */
 /*

